cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/transform/LogParserTransform.java

```diff
@@ -0,0 +1,220 @@
+  private static final Pattern REQUEST_PAGE_PATTERN = Pattern.compile("(\\S+)\\s(\\S+).*");
+  //Indices of which group request, time, ip, and user agent are in the S3 regex
+  private static final int[] S3_INDICES = {9, 3, 4, 17};
+  //Indices of which group request, time, ip, and user agent are in the CLF regex
+  private static final int[] CLF_INDICES = {5, 4, 1, 9};
+  //Number of groups matched in the S3 regex
+  private static final int S3_REGEX_LENGTH = 18;
+  //Number of groups matched in the CLF regex
+  private static final int CLF_REGEX_LENGTH = 9;
+  private static final String S3_LOG = "S3";
+  private static final String CLF_LOG = "CLF";
+  private final LogParserConfig config;
+  private final SimpleDateFormat sdf = new SimpleDateFormat("dd/MMM/yyyy:HH:mm:ss Z");
+
+  public LogParserTransform(LogParserConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) throws Exception {
+    String log = getLogFromInput(input);
+    if (log == null) {
+      LOG.error("Couldn't read schema, log message was null");
+      return;
+    }
+
+    StructuredRecord output = null;
+    if (!S3_LOG.equals(config.logFormat) && !CLF_LOG.equals(config.logFormat)) {
+      LOG.error("Log format not currently supported.");
+      return;
+    } else if (S3_LOG.equals(config.logFormat)) {
+      Matcher logMatcher = S3_LOG_PATTERN.matcher(log);
+      if (!logMatcher.matches() || logMatcher.groupCount() < S3_REGEX_LENGTH) {
+        LOG.debug("Couldn't parse log because log did not match the S3 format, log:" + log);
+        return;
+      }
+      output = parseRequest(logMatcher, S3_INDICES);
+    } else if (CLF_LOG.equals(config.logFormat)) {
+      Matcher logMatcher = CLF_LOG_PATTERN.matcher(log);
+      if (!logMatcher.matches() || logMatcher.groupCount() < CLF_REGEX_LENGTH) {
+        LOG.debug("Couldn't parse log because the log did not match the CLF format. log:" + log);
+        return;
+      }
+      output = parseRequest(logMatcher, CLF_INDICES);
+    }
+
+    if (output != null) {
+      emitter.emit(output);
+    }
+  }
+
+  /**
+   * Gets the log message from the input
+   * @param input the StructuredRecord to extract the log from
+   * @return the log message, or null on failure
+   */
+  @Nullable
+  private String getLogFromInput(StructuredRecord input) {
+    Schema.Field inputField = input.getSchema().getField(config.inputName);
+    if (inputField == null) {
+      LOG.debug("Invalid inputName, no known inputField matches given input");
+      return null;
+    }
+
+    Schema inputSchema = inputField.getSchema();
+    if (inputSchema.isNullableSimple()) {
+      inputSchema = inputSchema.getNonNullable();
+    }
+    Schema.Type inputType = inputSchema.getType();
+
+    if (!Schema.Type.STRING.equals(inputType) && !Schema.Type.BYTES.equals(inputType)) {
+      LOG.error("Unsupported inputType, only bytes and string are supported.");
```

Comment:
Also, good to mention here that  its "Unsupported type **in schema**"  (Also perhaps ``Schema.Type.BYTES`` (not just bytes) and ``Schema.Type.STRING`` (not just string))